### PR TITLE
Changes to remove Pip3 based installations for azdata

### PR DESCRIPTION
**Pull request by ranasaria**
This PR fixes the following:
1. Removes the default pip3 installations as we don't want to support them.
2. Moves the logic of user-defined settings based URL location for azdata installation from Pip3 based mechanism to apt-get/MSI/brew based installers.
3. Fixes an issue where the message: "Some required tools are not installed, you can click the 'Install Tools' button to install them"  was showing up even when the installation was already in progress.

Tested the following scenarios:
1. Normal positive case installation with 1, 2, or all 3 tools missing.
2. Faked in code to force the path when on installation commands are not available for an OsDistribution value of 'Others', to test out the user experience on those platforms.
[a] Tested with installation commands not available for azdata
[b] Tested with installation commands not available for 2 tools.

### CODE REVIEW

**Comment by alanrenmsft** [comment]
have you tested your changes on all platforms with all scenarios?

File: extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts [hunk 1]
```diff
@@ -358,14 +362,24 @@ export class ResourceTypePickerDialog extends DialogBase {
...
+					level: azdata.window.MessageLevel.Information,
+					text: localize('deploymentDialog.InstalledTools', "All required tools are installed now.")
+				};
+				this._dialogObject.okButton.enabled = true;
+			} else {
+				this._dialogObject.message = {
+					level: azdata.window.MessageLevel.Information,
+					text: localize('deploymentDialog.PendingInstallation', "Following tools: {0} were still not discovered. Please make sure that they are installed, running and discoverable", toolsNotInstalled.map(t => t.displayName).join(','))
+				};
+				//this.updateToolsDisplayTable(); // refresh the tools table data which will repaint the UI with current state of all tools.
```
Comment (alanrenmsft) [inline extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts]: remove if not needed

File: extensions/resource-deployment/src/services/tools/toolBase.ts [hunk 2]
```diff
@@ -27,17 +27,14 @@ const toolStatusLocalized: Map<ToolStatus, string> = new Map<ToolStatus, string>
 ]);
 
 export const enum dependencyType {
-	PythonAndPip3 = 'PythonAndPip3',
 	Brew = 'Brew',
```
Comment (aaomidi) [inline extensions/resource-deployment/src/services/tools/toolBase.ts]: Should the strings be lowercase?

File: extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts [hunk 3]
```diff
@@ -216,33 +218,33 @@ export class ResourceTypePickerDialog extends DialogBase {
...
-				this._toolsTable.data = toolRequirements.map(toolReq => {
-					const tool = this.toolsService.getToolByName(toolReq.name)!;
+				this._toolsTable.data = toolRequirements.map(toolRequirement => {
+					const tool = this.toolsService.getToolByName(toolRequirement.name)!;
 					// subscribe to onUpdateData event of the tool.
 					this._toDispose.push(tool.onDidUpdateData((t: ITool) => {
 						this.updateToolsDisplayTableData(t);
 					}));
 					if (tool.isNotInstalled && !tool.autoInstallSupported) {
-						messages.push(localize('deploymentDialog.ToolInformation', "'{0}' [ {1} ]", tool.displayName, tool.homePage));
```
Comment (alanrenmsft) [inline extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts]: '{0}' is either not installed or not running, You can find more information about the tool at: {1}

File: extensions/resource-deployment/src/services/tools/toolBase.ts (hunk 2 above)
Comment (ranasaria) [reply extensions/resource-deployment/src/services/tools/toolBase.ts]: It does not matter. They are not seen externally anywhere, but if you prefer, I can change them to lowercase. It was kept the same as enum value so it is easy to match the enum value.

File: extensions/resource-deployment/package.json [hunk 4]
```diff
@@ -29,15 +29,9 @@
...
+					"deployment.azdataInstallLocation": {
 						"type": "string",
-						"default": "https://aka.ms/azdata",
-						"description": "%azdata-pip-install-uri-description%"
-					},
-					"deployment.azdataPipInstallArgs": {
-						"type": "string",
-						"default": "",
-						"description": "%azdata-pip-install-args-description%"
+						"description": "%azdata-install-Location-description%"
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/package.json]: >L [](start = 38, length = 1)

nit: lowercase l

File: extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts [hunk 5]
```diff
@@ -358,14 +362,23 @@ export class ResourceTypePickerDialog extends DialogBase {
...
+				this._dialogObject.message = {
+					level: azdata.window.MessageLevel.Information,
+					text: localize('deploymentDialog.InstalledTools', "All required tools are installed now.")
+				};
+				this._dialogObject.okButton.enabled = true;
+			} else {
+				this._dialogObject.message = {
+					level: azdata.window.MessageLevel.Information,
+					text: localize('deploymentDialog.PendingInstallation', "Following tools: {0} were still not discovered. Please make sure that they are installed, running and discoverable", toolsNotInstalled.map(t => t.displayName).join(','))
+				};
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts]: This sentence could use some work - it doesn't read that well currently. At the very least something like `Unable to discover these tools: {0}...." would be better but I'd run this by PMs too

File: extensions/resource-deployment/src/services/tools/toolBase.ts (hunk 2 above)
Comment (aaomidi) [reply extensions/resource-deployment/src/services/tools/toolBase.ts]: Nah no big deal, was just curious if we use them like that

File: extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts (hunk 3 above)
Comment (ranasaria) [reply extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts]: Fixed.

**Comment by ranasaria** [comment]
> have you tested your changes on all platforms with all scenarios?

However, I have a couple of more fixes in the pipeline I will test them together on Mac/Linux. These changes are not platform-specific (the actual installation to switch azdata from pip3 to brew, apt-get, misexec was and that change was previously done and tested on all 3 platforms already.

**Pull request merged**